cachi2/core/package_managers/bundler/main.py

```diff
@@ -118,3 +125,35 @@ def _get_repo_name_from_origin_remote(package_dir: RootedPath) -> str:
 
     resolved_path = Path(repo_name).joinpath(package_dir.subpath_from_root)
     return str(resolved_path)
+
+
+def _prepare_environment_variables_for_hermetic_build() -> list[EnvironmentVariable]:
+    return [
+        # Contains path to a directory where a new config could be found.
+        EnvironmentVariable(name="BUNDLE_APP_CONFIG", value="${output_dir}/" + CONFIG_OVERRIDE),
+    ]
+
+
+def _prepare_for_hermetic_build(source_dir: RootedPath, output_dir: RootedPath) -> ProjectFile:
+    """Prepare a package for hermetic build by injecting necessary config."""
+    potential_bundle_config = source_dir.join_within_root(".bundle/config").path
+    hermetic_config = dedent(
+        """
+        BUNDLE_CACHE_PATH: "${output_dir}/deps/rubygems"
+        BUNDLE_DEPLOYMENT: "true"
+        BUNDLE_NO_PRUNE: "true"
+    """
+    )
+    if potential_bundle_config.is_file():
+        config_data = potential_bundle_config.read_text()
+        config_data += hermetic_config
+    # Corner case: a user decides to define their own alternate config.
+    # In this scenario cachi2 must try to copy over user-defined variables
+    # to its overriding alternate config.
+    elif (alternative_config := os.getenv("BUNDLE_APP_CONFIG")) is not None:
+        config_data = (Path(alternative_config) / "config").read_text()
```

Comment:
Here you can use `joinpath` and remove extra parentheses, but fine.

Reply:
`joinpath` should be IMO avoided, vast majority of cases can be written in a more compact way by just using `Path` directly. Note `joinpath` doesn't even work over the same object and returns a new one so there's no practical advantage. In this case `Path(alternative_config, "config") ` is all that's needed.

Reply:
Good point, done.